router/pkg/pubsub/datasource/datasource.go

```diff
@@ -0,0 +1,14 @@
+package datasource
+
+import (
+	nodev1 "github.com/wundergraph/cosmo/router/gen/proto/wg/cosmo/node/v1"
+	"github.com/wundergraph/graphql-go-tools/v2/pkg/engine/resolve"
+)
+
+type PubSubDataSource interface {
```

Comment:
Add extensive documentation on what this interface is, explain each func, add comprehensive info on how to implement such a PubSubDataSource

Reply:
Add a link to the README explaining how to implement a PubSubDS